OsStub/BaseCryptLibMbedTls/Hash/CryptSha2.c

```diff
@@ -0,0 +1,403 @@
+/** @file
+  SHA-2 Digest Wrapper Implementations over OpenSSL.
+
+  This file includes SHA256, SHA384, SHA512, all part of SHA-2 family.
+
+Copyright (c) 2009 - 2016, Intel Corporation. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include "InternalCryptLib.h"
+#include <mbedtls/sha256.h>
+#include <mbedtls/sha512.h>
+
+enum InternalSha2Type{_SHA2_256, _SHA2_384, _SHA2_512};
+
+/**
+
+  Computes the size of a mbedtls_sha{256, 512}_context struct given a
+  Sha2Type. Note that 384 and 512 both use mbedtls_sha512_context.
+
+  @param[in]   type     Algorithm requested. Must be supported.
+
+  @retval UINTN > 0   The size of the corresponding context struct.
+  @retval UINTN == 0   Error; algorithm not supported.
+
+**/
+static UINTN GetContextSize(enum InternalSha2Type type)
+{
+  switch (type)
+  {
+    case _SHA2_256:
+    {
+      return (UINTN)(sizeof(mbedtls_sha256_context));
+    }
+    case _SHA2_384:
+    case _SHA2_512:
+    {
+      return (UINTN)(sizeof(mbedtls_sha512_context));
+    }
+  }
+
+  return 0;
+}
+
+/**
+
+  Given a Sha2Type, allocate, initialize, and return a pointer to an
+  initialized context struct.
+
+  Note that this function will allocate.
+
+  @param[in]   type     Algorithm requested. Must be supported.
+
+  @retval VOID* != NULL An initialized context structure supporting algorithm `type`.
+  @retval VOID* == NULL Error; requested context could not be initialized.
+
+**/
+static VOID* InitAndStartShaContext(enum InternalSha2Type type)
+{
+  VOID* ctx;
+
+  ctx = (VOID*)AllocatePool(GetContextSize(type));
+
+  if (!ctx)
+    return FALSE;

```

Comment:
This coding style is not acceptable.

Please use :
if (!ctx) {
  return FALSE;
}